azure-quantum/tests/unit/test_job.py

```diff
@@ -96,48 +96,89 @@ def get_test_job_id(self):
 
     def test_job_submit_microsoft_simulated_annealing(self):
         solver_type = functools.partial(microsoft.SimulatedAnnealing, beta_start=0)
-        self._test_job_submit(solver_type)
+        solver_name = "SimulatedAnnealing"
+        self._test_job_submit(solver_name, solver_type)
         self._test_job_filter(solver_type)
 
     def test_job_submit_microsoft_parallel_tempering(self):
         solver_type = functools.partial(microsoft.ParallelTempering, sweeps=100)
-        self._test_job_submit(solver_type)
+        solver_name = "ParallelTempering"
+        self._test_job_submit(solver_name, solver_type)
 
     def test_job_submit_microsoft_tabu(self):
         solver_type = functools.partial(microsoft.Tabu, sweeps=100)
-        self._test_job_submit(solver_type)
+        solver_name = "Tabu"
+        self._test_job_submit(solver_name, solver_type)
 
     def test_job_submit_microsoft_quantum_monte_carlo(self):
         solver_type = functools.partial(microsoft.QuantumMonteCarlo, trotter_number=1)
-        self._test_job_submit(solver_type)
+        solver_name = "QuantumMonteCarlo"
+        self._test_job_submit(solver_name, solver_type)
 
     def test_job_submit_microsoft_population_annealing(self):
         solver_type = functools.partial(microsoft.PopulationAnnealing, sweeps=200)
-        self._test_job_submit(solver_type)
+        solver_name = "PopulationAnnealing"
+        self._test_job_submit(solver_name, solver_type)
 
     def test_job_submit_microsoft_substochastic_monte_carlo(self):
         solver_type = functools.partial(microsoft.SubstochasticMonteCarlo, step_limit=280)
-        self._test_job_submit(solver_type)
+        solver_name = "SubstochasticMonteCarlo"
+        self._test_job_submit(solver_name, solver_type)
+
+    def test_job_upload_and_run_solvers(self):
+        problem_name = f'Test-problem-{datetime.now():"%Y%m%d-%H%M%S"}'
+        problem = self.create_problem(name=problem_name)
+        workspace = self.create_workspace()
+
+        with unittest.mock.patch.object(
+            Job,
+            self.mock_create_job_id_name,
+            return_value=self.get_test_job_id(),
+        ):
+            workspace = self.create_workspace()
+            blob = problem.to_blob(compress=False)
+            job = Job.from_input_data(
```

Comment:
I wouldn't use `Job.from_input_data` as the mechanism to upload input_data if  you want to use it on multiple jobs.  I would use the `problem.upload` method to upload the Problem, then I would move creating the job instead to the `for` loop below as you really want to create one Job per solver.

Reply:
Good call, thanks!